### DISCUSSION TITLE
Using proxy in replication

### DISCUSSION THREAD

**Discussion by inadreev** [tarantool/tarantool#7795] [Q&A] [answered]
Good day,
I have a technical question about the work of your DBMS.
It is possible to perform the proxying of traffic using it? Traffic that should be sent between replicas in Tarantool cluster.
I mean a service channel between replicas (port 3301). Is it possible to send this traffic through SOCKS5 or HTTP proxy?

**Comment by Totktonada** [comment, marked as answer]
Tarantool binary protocol (iproto) is NOT based on HTTP(s). It is a TCP traffic with TLS inside[^1] (TLS is optional) with msgpack encoded requests and responses. So one need a TCP proxy for proxying this traffic, not an HTTP(s) proxy. I know setups with HAProxy between a connector (iproto client) and a tarantool server. I don't know about setups with HAProxy in between replication peers, but logically it shouldn't break things.

Just curious, why a proxy is needed here?

[^1]: Available in the Enterprise Edition.
